aodntools/timeseries_products/aggregated_timeseries.py

```diff
@@ -319,231 +235,201 @@ def source_file_attributes(download_url_prefix, opendap_url_prefix):
-                                                          'LATITUDE': nc.LATITUDE.squeeze().values,
-                                                          'NOMINAL_DEPTH': get_nominal_depth(nc)}, ignore_index = True)
-                    fileIndex += 1
-                else:
-                    rejected_files.append(file)
-                    bad_files.update({file: file_problems})
-        except:
-            print("FILE NOT FOUND: " + file, file=sys.stderr)
-
-    print()
-
-
-    ## rename indices
-    variableAuxDF.index.rename('INSTRUMENT', inplace=True)
-    variableMainDF.index.rename('OBSERVATION', inplace=True)
-
-    ## get the list of variables
-    varlist = list(variableMainDF.columns) + list(variableAuxDF.columns)
-
-    ## set variable attributes
-    add_variable_attribute = {'PRES_REL': {'applied_offset_by_instrument': applied_offset}}
+        with xr.open_dataset(os.path.join(input_dir, file)) as nc:
+            ## clip to in water data only
+            nc = in_water(nc)
+
+            error_list = check_file(nc, site_code, var_to_agg)
+            if not error_list:
+                varlen_list.append(len(nc.TIME))
+            else:
+                bad_files.update({file: error_list})
+                rejected_files.append(file)
+
+
+    ## remove bad files form the list
+    for file in bad_files.keys():
+        files_to_agg.remove(file)
+
+
+    varlen_list = [0] + varlen_list
+    varlen_total = sum(varlen_list)
+    n_files = len(files_to_agg)
+
+    ## create ncdf file, dimensions and variables
+    ds = Dataset(os.path.join(output_dir, outfile), 'w')
+    OBSERVATION = ds.createDimension('OBSERVATION', size=varlen_total)
+    INSTRUMENT = ds.createDimension('INSTRUMENT', size=n_files)
+
+    obs_float_template = {'datatype': 'float', 'zlib': True, 'dimensions': ('OBSERVATION'), "fill_value": 99999.0}
+    obs_byte_template = {'datatype': 'byte', 'zlib': True, 'dimensions': ('OBSERVATION'), 'fill_value': 99}
+    obs_int_template = {'datatype': 'int', 'zlib': True, 'dimensions': ('OBSERVATION')}
+    inst_S256_template = {'datatype': 'S256', 'dimensions': ('INSTRUMENT')}
+    inst_float_template ={'datatype': 'float', 'dimensions': ('INSTRUMENT'), "fill_value": 99999.0}
+
+    agg_variable = ds.createVariable(varname=var_to_agg, **obs_float_template)
+    agg_variable_qc = ds.createVariable(varname=var_to_agg + '_quality_control', **obs_byte_template)
+    DEPTH = ds.createVariable(varname='DEPTH', **obs_float_template)
+    DEPTHqc = ds.createVariable(varname='DEPTH_quality_control', **obs_byte_template)
+    PRES = ds.createVariable(varname='PRES', **obs_float_template)
+    PRESqc = ds.createVariable(varname='PRES_quality_control', **obs_byte_template)
+    PRES_REL = ds.createVariable(varname='PRES_REL', **obs_float_template)
+    PRES_RELqc = ds.createVariable(varname='PRES_REL_quality_control', **obs_byte_template)
+
+    TIME = ds.createVariable(varname='TIME', **obs_float_template)
```

Comment:
TIME, LATITUDE, LONGITUDE _should_ be of type `double`

Reply:
fixed. Also, the other dtypes set to float32